### PR TITLE
Replace context.queries.primaryShopId shopId with client shopId

### LINKED ISSUE #5801: [3.0.0] Fix improper use of context.queries.primaryShopId
context.queries.primaryShopId is used in several places. These need to be updated to have client pass in a shopId.

### PR DESCRIPTION
**Pull request by trojanh**
Resolves #5801 
Impact: **minor**  
Type: **refactor**

## Issue
context.queries.primaryShopId is used in several places. These need to be updated to have client pass in a shopId.

## Solution
Remove `context.queries.primaryShopId` from following mutations/queries:

- [x] [publishNavigationChanges](https://github.com//reactioncommerce/reaction/blob/dba6fa1980f06f04a4d17afbf06ddb1fcfbccda9/src/plugins/navigation/mutations/publishNavigationChanges.js#L17)
- [x] [updateNavigationItem](https://github.com//reactioncommerce/reaction/blob/d2080cfdf7527de36851592e1d96d40feb52fd19/src/plugins/navigation/mutations/updateNavigationItem.js#L17)
- [x] [updateNavigationTree](https://github.com//reactioncommerce/reaction/blob/d2080cfdf7527de36851592e1d96d40feb52fd19/src/plugins/navigation/mutations/updateNavigationTree.js#L18)
- [x] [catalog/queries/tag](https://github.com//reactioncommerce/reaction/blob/fd09b9e88d0c6e23fcaba2bae3701b66ed09ee86/src/core-services/catalog/queries/tag.js) 
- [x] [generateSitemaps](https://github.com//reactioncommerce/reaction/blob/d2080cfdf7527de36851592e1d96d40feb52fd19/src/plugins/sitemap-generator/mutations/generateSitemaps.js#L12)

- [x] Add test for tag query and api

### CODE REVIEW

**Comment by trojanh** [comment]
@aldeed could you verify if I have covered all the places?

**Comment by trojanh** [comment]
@aldeed [catalog/queries/tag](https://github.com//reactioncommerce/reaction/blob/fd09b9e88d0c6e23fcaba2bae3701b66ed09ee86/src/core-services/catalog/queries/tag.js)  doesn't have graphql schema query, is it meant for internal use or we need to create its endpoint later?

**Comment by trojanh** [comment]
Please ignore last my comment. I have figured that it was used in the tags plugin [here](https://github.com//reactioncommerce/reaction/blob/1b8993fdc37afc247a02dd651c48e1a45911f0fa/src/core-services/tags/resolvers/Query/tag.js#L35)

File: src/plugins/navigation/mutations/publishNavigationChanges.js [hunk 1]
```diff
@@ -6,20 +6,20 @@ import getNavigationTreeItemIds from "../util/getNavigationTreeItemIds.js";
  * @method publishNavigationChanges
  * @summary Publishes changes for a navigation tree and its items
  * @param {Object} context An object containing the per-request state
- * @param {String} _id _id of navigation tree to publish
+ * @param {String} input input of navigation tree to publish
+ * @param {String} shopId shopId of navigation tree to publish
  * @returns {Promise<Object>} Updated navigation tree
  */
-export default async function publishNavigationChanges(context, _id) {
+export default async function publishNavigationChanges(context, input) {
```
Comment (aldeed) [inline src/plugins/navigation/mutations/publishNavigationChanges.js]: This is called internally by `src/plugins/navigation/util/createDefaultNavigationTreeForShop.js`. Need to update the function signature there.

File: src/plugins/navigation/resolvers/Mutation/publishNavigationChanges.js [hunk 2]
```diff
@@ -15,12 +15,16 @@ import { decodeNavigationTreeOpaqueId } from "../../xforms/id.js";
...
+    _id,
+    shopId: opaqueShopId
   } = input;
 
   const decodedId = decodeNavigationTreeOpaqueId(_id);
-
-  const publishedNavigationTree = context.mutations.publishNavigationChanges(context, decodedId);
+  const shopId = decodeShopOpaqueId(opaqueShopId);
+  const publishedNavigationTree = context.mutations.publishNavigationChanges(context, {
+    decodedId,
```
Comment (aldeed) [inline src/plugins/navigation/resolvers/Mutation/publishNavigationChanges.js]: Should be `_id: decodedId`

**Comment by aldeed** [comment]
@trojanh There is one more place you missed: the `generateSitemaps` mutation in the `sitemap-generator` plugin. It should be similar to the others. Add `shopId` to gql and pass along.

**Comment by trojanh** [comment]
@aldeed I have made the suggested change and also observed a few of following inconsistencies which I have fixed:
 - Most API input takes `id` to fetch any resource, in few of the above we were using `_id`, so I have made it consistent with other naming conventions
 - Fix documentation for some of the mutations and schemas
 - Add test cases for `tag` query and API

**Pull request merged**